refactor: move telemetry interfaces to core interfaces (#16211)
microsoft/FluidFramework: common/lib/common-definitions/CHANGELOG.md

## [1.2.0](https://github.com/microsoft/FluidFramework/releases/tag/common-definitions_v1.2.0)

### Deprecated interfaces and types

> The following interfaces and types are deprecated in this release. Its replacement has been moved to another package.

Reviewer: Nit
```suggestion
The following interfaces and types are deprecated in this release. Their replacements have been moved to another package.
```
